### ISSUE TITLE
Already-loaded login passwords kept available even after keychain is re-locked

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jfitzell/mozilla-keychain#5]
```
First, thanks for the time you've put into this - it is very much appreciated.

I believe I see some incorrect behavior with this extension when it comes
to what Firefox refers to as active logins on password-protected webpages
(protected w/ .htaccess). If I've authenticated to a page, then locked my
keychain - say by starting the screensaver or putting the computer to sleep
- I would argue that those active logins should be cleared and keychain
reauthentication be required before I am allowed to reload that page or
access any other page sharing that set of restrictions. Instead, the active
login continues, and I can access any pages protected by that particular
login without having to access the keychain again.

What steps will reproduce the problem?
1. Log in to a password protected page or site (.htaccess type restrictions
requiring a username and password)
2. Lock the keychain.
3. Force a page reload - no reauthentication will be required

What is the expected output? What do you see instead?

I expected I'd have to type in my keychain password before being allowed to
proceed. Instead, I still had access to the restricted page.

What version of the product are you using? On what operating system?

1.0a7 running in Firefox 3.5.2 on OS X 10.5.8

Please provide any additional information below.

While I would argue this is a bug because the behavior is not as expected,
in terms of coding I think it's technically an enhancement request - adding
code to clear existing "active logins" if the keychain is currently locked,
in other words.


```

Original issue reported on code.google.com by `tsal...@gmail.com` on 21 Aug 2009 at 5:20
